### DISCUSSION TITLE
How to get Timing information if I use the Node.js API?

### DISCUSSION THREAD

**Discussion by Mattstir** [eslint/eslint#16495] [Q&A] [answered]
Is there a way to get the [TIMING information ](https://eslint.org/docs/latest/developer-guide/working-with-rules#per-rule-performance) if I use the [Node.js API](https://eslint.org/docs/latest/developer-guide/nodejs-api)?

I can't spot any reference to this in the Docs.

Thanks! :)

**Comment by mdjermanovic** [comment, marked as answer]
The TIMING feature was designed for CLI: it [`console.log`](https://github.com/eslint/eslint/blob/c97c7897686ac4dc2828537d6a017f3c99f7d905/lib/linter/timing.js#L119)s the timing info [on process exit](https://github.com/eslint/eslint/blob/c97c7897686ac4dc2828537d6a017f3c99f7d905/lib/linter/timing.js#L150-L152). There's no way to get this information through the API.

**Reply by Mattstir** [reply]
Thanks for the info, just as expected after not finding anything in the Docs.

Are there plans to add it? Maybe in a slightly different way. Setting a options prop when creating a new Eslint Object, where you can get it from after the lint run.
We would heavily benefit from this possible feature. Thanks! :)

**Reply by mdjermanovic** [reply]
We currently have no plans to add this feature to the API. If you open a change request describing your use case, we could certainly discuss it.

**Reply by Mattstir** [reply]
Created https://github.com/eslint/eslint/issues/16521
